.travis.yml

```diff
@@ -18,25 +25,28 @@ before_install:
      tesseract-ocr
      tesseract-ocr-deu
      unzip
-  - wget https://johnvansickle.com/ffmpeg/releases/ffmpeg-release-64bit-static.tar.xz
-  - tar xf ffmpeg-release-64bit-static.tar.xz
-  - sudo mv ffmpeg-*/ff* /usr/bin/
+  - _nt docs wget https://johnvansickle.com/ffmpeg/releases/ffmpeg-release-64bit-static.tar.xz
+  - _nt docs tar xf ffmpeg-release-64bit-static.tar.xz
+  - _nt docs sudo mv ffmpeg-*/ff* /usr/bin/
   - >
+     _nt docs
      wget https://mvncache.opencast.org/nexus/content/repositories/thirdparty/org/synfig/synfigstudio/1.0.2/synfigstudio-1.0.2.deb ||
      wget https://nexus.opencast.org/nexus/content/repositories/thirdparty/org/synfig/synfigstudio/1.0.2/synfigstudio-1.0.2.deb ||
      wget -O synfigstudio-1.0.2.deb https://netix.dl.sourceforge.net/project/synfig/releases/1.0.2/linux/synfigstudio_1.0.2_amd64.deb
-  - sudo dpkg -i synfigstudio-1.0.2.deb
-  - sudo pip install mkdocs
+  - _nt docs sudo dpkg -i synfigstudio-1.0.2.deb
+  - _t docs sudo pip install mkdocs
 
 install:
   - true
 
 script:
-  - (! grep -rn '	' modules assemblies pom.xml --include=pom.xml)
-  - (! grep -rn ' $' modules assemblies pom.xml --include=pom.xml)
-  - (! grep -rn '	' etc)
-  - (! grep -rn ' $' etc)
-  - (cd docs/guides/admin && mkdocs build)
-  - (cd docs/guides/developer && mkdocs build)
-  - (cd docs/guides/user && mkdocs build)
-  - mvn --batch-mode clean install -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -Pnone
+  - _t docs sh -c "! grep -rn '	' modules assemblies pom.xml --include=pom.xml"
+  - _t docs sh -c "! grep -rn ' $' modules assemblies pom.xml --include=pom.xml"
+  - _t docs sh -c "! grep -rn '	' etc"
+  - _t docs sh -c "! grep -rn ' $' etc"
+  - _t docs sh -c "cd docs/guides/admin && mkdocs build"
+  - _t docs sh -c "cd docs/guides/developer && mkdocs build"
+  - _t docs sh -c "cd docs/guides/user && mkdocs build"
+  - _t build mvn --batch-mode clean install -Pnone -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
+  - _t site  mvn --batch-mode install -DskipTests=true -Pnone -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
```

Comment:
What's up with this second run?  Why isn't the first one enough?

Reply:
Hm, maybe, I should have explained this a bit more. `_t` will execute the following command only if the `T` environment variable is set to the first argument of this method.

The build matrix specified here will launch three parallel builds with `T` set to docs, build and site.

- The docs-build will just build our documentation and do some simple checks on our configuration and pom.xml files (no trailing spaces or tabs).
- The build-build will do a regular build with all tests enabled (line 50).
- The site-build will first do a quick build with all tests disabled and then run `mvn site`. The initial build is necessary since `mvn site` will fail otherwise.

Reply:
Ah, I missed the initial build docs site mechanism.  Looks good to me!